includes/ziti/ziti_events.h

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (c) 2020 NetFoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+#ifndef ZITI_SDK_ZITI_EVENTS_H
+#define ZITI_SDK_ZITI_EVENTS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    ZitiContextEvent = 1,
+    ZitiRouterEvent = 1 << 1,
+    ZitiServiceEvent = 1 << 2,
+} ziti_event_type;
+
+typedef enum {
+    EdgeRouterConnected,
+    EdgeRouterDisconnected,
+    EdgeRouterRemoved,
+    EdgeRouterUnavailable,
+} ziti_router_status;
+
+struct ziti_context_event {
+    int ctrl_status;
```

Comment:
just an `int`? (no type like you see for e.g., `ziti_router_status`)

Reply:
should it be renamed to `ziti_ctrl_event` for consistency then too? i like the idea of "router events" and "controller" events in that way. there's also "service events" so it's consistent